File: knowledge_base/springboot/demo/src/main/java/com/kennybaik/demo/service/UserService.java
package com.kennybaik.demo.service;

import com.kennybaik.demo.entity.User;
import com.kennybaik.demo.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService {

    private final UserRepository userRepository;

    @Autowired
    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    // 모든 사용자 조회
    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    // 특정 사용자 조회
    public Optional<User> getUserById(Long id) {
        return userRepository.findById(id);
    }

    // 사용자 추가
    public User createUser(User user) {
        return userRepository.save(user);
    }

    // 사용자 삭제
    public void deleteUser(Long id) {
        userRepository.deleteById(id);
    }
}

File: knowledge_base/springboot/docker+mysql+aws+redis/src/main/java/com/example/demo/service/UserService.kt
package com.example.springbootapp.service

import com.example.springbootapp.entity.User
import com.example.springbootapp.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService(private val userRepository: UserRepository) {

    // 이메일로 사용자 조회
    fun getUserByEmail(email: String): User? {
        return userRepository.findByEmail(email).orElse(null)
    }

    // 사용자 생성
    fun createUser(user: User): User {
        return userRepository.save(user)
    }

    // 모든 사용자 조회
    fun getAllUsers(): List<User> {
        return userRepository.findAll()
    }
}
